server/store/store.go

```
@@ -37,7 +43,22 @@ func openAdapter(workerId int, jsonconf json.RawMessage) error {
 	}
 
 	if adp == nil {
-		return errors.New("store: database adapter is missing")
+		if len(config.AdapterName) > 0 {
+			// Adapter name specified explicitly.
+			if ad, ok := availableAdapters[config.AdapterName]; ok {
+				adp = ad
+			} else {
+				return errors.New("store: " + config.AdapterName + " adapter is not available in this binary")
+			}
+		} else if len(availableAdapters) == 1 {
+			// Attempt to default to the only entry in availableAdapters.
+			for _, v := range availableAdapters {
+				adp = v
+			}
+			log.Println("store: db adapter not specified explicitly; using", adp.GetName())
```

I'm not sure this is a good idea. Let it fail if misconfigured.